interface/src/app/sample/sampleDetail.component.js

```diff
@@ -0,0 +1,71 @@
+angular.module('adage.sampleDetail', [
+  'adage.sample.service',
+  'adage.experiment.service',
+  'statusBar'
+])
+
+.component('sampleDetail', {
+  templateUrl: 'sample/sampleDetail.component.tpl.html',
+  bindings: {
+    id: '<',
+    onLoad: '&'
+  },
+  controller: [
+    'Sample', 'Experiment', '$log',
+    function(Sample, Experiment, $log) {
+      var ctrl = this;
+      ctrl.makeHref = Experiment.makeHref;
+
+      ctrl.$onChanges = function(changesObj) {
+        ctrl.show(changesObj.id.currentValue);
+      };
+      ctrl.sample = {
+        status: '',
+        results: {},
+        relatedExperiments: []
+      };
+
+      var queryError = function(responseObject) {
+        $log.warn(
+          'adage.sampleDetail: query errored with: ' + responseObject
+        );
+        ctrl.sample.status = 'Query failed.';
+      };
+
+      ctrl.show = function(id) {
+        if (!id) {
+          $log.warn('adage.sampleDetail: show() called with id', id);
+          return;
+        }
+        ctrl.sample = {
```

Comment:
Oooh. Good catch! What you describe is the way I did it for experiment. Not sure why I missed that here. Will fix.

- [ ] change how we update `ctrl.sample` in `sampleDetail.component.js`

Reply:
Oh, no. Actually, the experiment works this way too and it should work this way. The outer initialization ensures those properties are defined as soon as the component is loaded. The inner one here ensures those properties are re-initialized any time the `ctrl.show(id)` method is called. This is important in case a parent component wants to call `.show` multiple times without fully reloading the `sampleDetail` component. So, this logic should not be changed.

Does that makes sense?

Reply:
Yes, that makes sense. Thanks. I guess if we want to avoid the duplicated code, we could create a method like `init()`, but let's keep it as is for now.

Reply:
I was just realizing that you had noticed this by the duplicated code. Making an `init()` is a good solution to that. Will do.

- [ ] reduce duplicated initialization code by pulling it out into a method

Reply:
This was a good place to use the component's `$onInit()` lifecycle hook, so that's how I fixed it:

> https://docs.angularjs.org/guide/component

Let me know what you think!